.github/workflows/release-contracts.yml

```diff
@@ -0,0 +1,51 @@
+name: Release contracts
+
+permissions:
+  contents: write
+
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - main
+      - ci/release-contracts
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    container: cosmwasm/workspace-optimizer:0.12.11
+    steps:
+      - uses: actions/checkout@v3
+
+      # tar is required for cargo cache
+      - run: apk add --no-cache tar
+
+      - name: Set up cargo cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/            
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Compile contracts
+        timeout-minutes: 30
+        run: optimize_workspace.sh .
+
+      - name: Upload contracts
+        uses: actions/upload-artifact@v3
+        with:
+          name: contracts
+          path: artifacts/
+
+      - name: release
+        uses: softprops/action-gh-release@v1
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          files: artifacts/*
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
@albttx, could you help me understand this line? from reading [the package](https://github.com/softprops/action-gh-release) it seems like this will automatically create releases, or will I still be able to cut them manually? i like to do a manual check of the checksums before releasing so I'm not sure automatically creating releases when tags are pushed is what we want.

Reply:
Hey,
If you want to see an example take a look at my fork https://github.com/albttx/dao-contracts/releases/tag/v4.1.0-test

I gave you permission, so try to generate a new release and you'll see the wasm contracts + checksum will be automatically published into the release.

Yes, every push of a git tag will automatically generate a release, but you can also just generate a release like you used to do! Try on my fork, you'll see, it's life changing! :)